### ISSUE TITLE
rust-no-float-for-money: false positive on 'balance_factor: f32' in K-Means clustering (cluster-balance weight, not monetary)

### ISSUE THREAD

**Issue by rbaumier** [rbaumier/comply#5598]
**Project**: lancedb/lance

**File**: `rust/lance-index/src/vector/kmeans.rs:80,148,256,277,310,344,486` (7 hits)

```rust
/// Balance factor for the kmeans clustering.
/// Higher value means more balanced clustering.
///
/// Setting this value to 0 means no balance factor,
/// which is the same as normal kmeans clustering.
pub balance_factor: f32,
```

And:
```rust
fn compute_balance_loss(cluster_sizes: &[usize], n: usize, balance_factor: f32) -> f32 {
    let size_loss = cluster_sizes.iter().map(|size| size.pow(2)).sum::<usize>() as f32;
    balance_factor * (size_loss - n.pow(2) as f32 / cluster_sizes.len() as f32)
}
```

comply reports: `rust-no-float-for-money: 'balance_factor: f32' — money values in floats accumulate IEEE 754 rounding errors. Use rust_decimal::Decimal.`

**Why it's a false positive**: `balance_factor` is a K-Means clustering tuning parameter that controls how evenly points are distributed across clusters (a standard ML hyperparameter, sometimes called a "regularization factor"). The word "balance" here refers to statistical cluster balance (uniform distribution of data points), not a financial balance (account balance, payment balance). Using `Decimal` for a clustering weight coefficient would be incorrect — IEEE 754 `f32` is the standard type for ML hyperparameters and loss function coefficients.

The rule includes `balance` as a monetary keyword, but the compound `balance_factor` is unambiguously ML/statistical (the word "factor" disambiguates). Similar precedent: issue #4748 found that `total_weight` is not a monetary quantity when used in graph algorithms.

**Expected**: `rust-no-float-for-money` should not flag `balance_factor` (or `balance_*` where the suffix is a domain-clarifying noun like `factor`, `loss`, `coefficient`). Alternatively, `balance` should only match as a standalone token (`field == "balance"`) or with monetary suffixes (`_balance`), not as a prefix in domain-qualified compounds.
